### PR TITLE
Update to Terminus 2

### PR DESCRIPTION
**Pull request by greg-1-anderson**

### CODE REVIEW

**Comment by ataylorme** [comment]
I'll need to either test manually, write some tests, or both

**Comment by greg-1-anderson** [comment]
The functional tests in the terminus-plugin-example use BATS to test using simple shell scripts. BATS probably isn't even necessary - just running a script with`set -e` would also be pretty simple and effective.

**Comment by davidneedham** [comment]
This works for me in Terminus 2.x as part of @ataylorme 's [Bulk Site Create/Clone script](https://github.com/ataylorme/pantheon-bulks-site-create-clone).

**Comment by ataylorme** [comment]
@greg-1-anderson what do you think about doing a `2.0.0` version of `terminus-site-clone` with Terminus `2` support and deprecating the current `1.0.0` release?

**Comment by greg-1-anderson** [comment]
@ataylorme The site clone plugin is mostly going to be used on the end-user's development machine. Most users should upgrade to Terminus 2 pretty quickly; I don't think that ongoing support for Terminus 1 is critical.

For Build Tools, I left the 1.x branch compatible with Terminus 1 (only), and I made the 2.x / master branch compatible with Terminus 2 (only).  This is certainly the easiest thing for plugins that make any API calls that vary from one major version of Terminus to the next.

**Comment by vinmassaro** [comment]
Subscribing to this issue and can do testing of any PRs since we used this plugin often with Terminus 1.x. Thanks!

**Comment by ataylorme** [comment]
@vinmassaro can you test the update I just pushed?

File: src/Commands/SiteCloneCommand.php [hunk 1]
```diff
@@ -522,7 +517,12 @@ private function wpcliSearchReplace()
             ]
         );
 
-        $this->destination['env_raw']->sendCommandViaSsh('wp search-replace ' . $this->source['pantheon_domain'] . ' ' . $this->destination['pantheon_domain']);
+        $WPCommand = new WPCommand();
+
+        $WPCommand->wpCommand(
```
Comment (ataylorme) [inline src/Commands/SiteCloneCommand.php]: I couldn't get [`wpCommand`](https://github.com/pantheon-systems/terminus/blob/master/src/Commands/Remote/WPCommand.php#L34) to work - it kept failing with the following error:

```sh
Using wp-cli search-replace to replace dev-wp-quicksilver-demo.pantheonsite.io with clone-test-december-couch-coding.pantheonsite.io on b28f9ea5-a2d7-445b-b047-2c174a276205.clone-test as WordPress stores URLs in the database. You may need to run the search-replace manually if you are using custom domains.

PHP Fatal error:  Uncaught Error: Call to a member function get() on null in /Users/andrewtaylor/pantheon-cli/src/Site/SiteAwareTrait.php:46
Stack trace:
#0 /Users/andrewtaylor/pantheon-cli/src/Site/SiteAwareTrait.php(87): Pantheon\Terminus\Commands\Remote\SSHBaseCommand->getSite('december-couch-...')
#1 /Users/andrewtaylor/pantheon-cli/src/Commands/Remote/SSHBaseCommand.php(49): Pantheon\Terminus\Commands\Remote\SSHBaseCommand->getSiteEnv('december-couch-...')
#2 /Users/andrewtaylor/pantheon-cli/src/Commands/Remote/WPCommand.php(36): Pantheon\Terminus\Commands\Remote\SSHBaseCommand->prepareEnvironment('december-couch-...')
#3 /Users/andrewtaylor/.terminus/plugins/terminus-site-clone-plugin/src/Commands/SiteCloneCommand.php(523): Pantheon\Terminus\Commands\Remote\WPCommand->wpCommand('december-couch-...', Array, Array)
#4 /Users/andrewtaylor/.terminus/plugins/terminus-site-clone-plugin/src/Commands/SiteCloneCommand.php(387): Pantheon\TerminusSiteClone\Commands\SiteCloneCommand->wpcliSearchReplace()
#5 /Users/andrewtaylor/.term in /Users/andrewtaylor/pantheon-cli/src/Site/SiteAwareTrait.php on line 46
```

**Comment by vinmassaro** [comment]
@ataylorme :100:
Was able to successfully clone a D7 site on the first try. The site files were also around ~1.13GB and I did not encounter any issues as I would have expected based on the docs warning about the 500MB limit.

**Comment by ataylorme** [comment]
@vinmassaro glad the Terminus 2 version in this PR is working for you! I am going to leave this open for a bit to allow for more testing and feedback before merging and tagging a new release.

**Pull request merged**